lightningd/onchain_control.c

```diff
@@ -518,6 +518,43 @@ static u32 infinite_block_deadline(const struct chain_topology *topo)
 	return get_block_height(topo) + 300;
 }
 
+/**
+ * slow_sweep_dealine -- A smart(-ish) way of estimating sweep fees.
+ *
+ * We use the `close_blockheight` as the basis for the target. We then
+ * target to have the funds back at most 2 weeks after the close,
+ * linearly increasing the confirmation target until we get
+ * close. However, we also don't want to panic and set a target that
+ * is too close, in order not to waste too many funds on the sweep
+ * fees.
+ */
+static u32 slow_sweep_deadline(const struct chain_topology *topo, const struct channel *c)
+{
+	u32 deadline, height = get_block_height(topo);
+	u32 closeheight = *c->close_blockheight ? *c->close_blockheight : 0;
```

Comment:
This test should be `c->close_blockheight ?` not  `*c->close_blockheight ?`.

And it should never happen (onchaind gets fired by code which sets the close_blockheight!) , so let's rework:

```
if (!c->close_blockheight) {
   log_broken(c->log, "Missing close_blockheight?  Using current height + 300!");
   return get_block_height(topo) + 300;
}
```

Reply:
Yep, my bad, the check should not dereference. Made it clearer by adding an `if` statement to disambiguate.